src/lib/Components/Consignments/Screens/Metadata.tsx

```diff
@@ -0,0 +1,154 @@
+  updateDepth = depth => this.setState({ depth })
+
+  animateStateChange = newState => {
+    const animate = LayoutAnimation.easeInEaseOut as any
+    animate()
+    this.setState(newState)
+  }
+
+  showCategorySelection = () => this.animateStateChange({ showSelector: true })
+  hideCategorySelection = () => this.animateStateChange({ showSelector: false })
+
+  render() {
+    return (
+      <ConsignmentBG>
+        <DoneButton onPress={this.doneTapped}>
+          <ScrollView>
+            <View>
+              <Row>
+                <Text
+                  text={{ placeholder: "Title", onChangeText: this.updateTitle, value: this.state.title }}
+                  style={{ margin: 10 }}
+                />
+              </Row>
+
+              <Row>
+                <Text
+                  text={{ placeholder: "Year", onChangeText: this.updateYear, value: this.state.year }}
+                  style={{ margin: 10 }}
+                />
+              </Row>
+
+              <Row>
+                <Text
+                  text={{
+                    placeholder: "Category",
+                    onChangeText: this.updateCategory,
+                    onFocus: this.showCategorySelection,
+                    value: this.state.category,
+                  }}
+                  style={{ margin: 10 }}
+                />
+              </Row>
+
+              {this.state.showSelector
+                ? <Picker
+                    style={{ backgroundColor: "white" }}
+                    selectedValue={this.state.category}
+                    onValueChange={(itemValue, itemIndex) => this.setState({ category: itemValue })}
+                  >
+                    {categoryOptions.map(category => <Picker.Item label={category} value={category} />)}
+                  </Picker>
+                : null}
+
+              <Row>
+                <Text
+                  text={{ placeholder: "Materials", onChangeText: this.updateMaterials, value: this.state.materials }}
+                  style={{ margin: 10 }}
+                />
+              </Row>
+
+              <Row>
+                <Text
+                  text={{ placeholder: "Width", onChangeText: this.updateWidth, value: this.state.width }}
+                  style={{ margin: 10 }}
+                />
+                <Text
+                  text={{
+                    placeholder: "Height",
+                    onChangeText: this.updateHeight,
+                    value: this.state.height,
+                  }}
+                  style={{ margin: 10 }}
+                />
+              </Row>
+
+              <Row>
+                <Text text={{ placeholder: "Depth" }} style={{ margin: 10 }} />
+                <View style={{ flex: 1, flexDirection: "row", alignContent: "center", margin: 10 }}>
+                  <Label>Units</Label>
+                  <Toggle selected={this.state.unit === "cm"} left="CM" right="IN" onPress={this.updateSigned} />
```

Comment:
You may be planning on some later transformation, but in convection this field is called `dimensions_metric` in case you want to keep the prop consistent here.